drivers/sensor/tsl2520/tsl2520.c

```diff
@@ -0,0 +1,220 @@
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2023 Carl Zeiss Meditec AG
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT ams_tsl2520
+
+#include "tsl2520.h"
+
+#include <stdlib.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/pm/device.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/pm/device.h>
+#include <zephyr/sys/__assert.h>
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/init.h>
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/sys/util.h>
+
+#define TSL2520_CHIP_ID 0x5C
+
+#define LO_BYTE(X)	(X & 0xFF)
+#define HI_BYTE(X)	((X>>8) & 0xFF)
+
+LOG_MODULE_REGISTER(tsl2520, CONFIG_SENSOR_LOG_LEVEL);
+
+static int tsl2520_sample_fetch(const struct device *dev, enum sensor_channel chan)
+{
+	const struct tsl2520_config *cfg = dev->config;
+	struct tsl2520_data *data = dev->data;
+	int ret = 0;
+
+	__ASSERT_NO_MSG(chan == SENSOR_CHAN_ALL || chan == SENSOR_CHAN_LIGHT ||
+			chan == SENSOR_CHAN_IR);
+
+	if (chan == SENSOR_CHAN_ALL || chan == SENSOR_CHAN_LIGHT || chan == SENSOR_CHAN_IR) {
+		uint8_t read_low_byte;
+		uint8_t read_high_byte;
+		uint8_t reg_status;
+
+		/* In order to update ALS Data Registers ALS_STATUS must be read first */
+		(void)i2c_reg_read_byte_dt(&cfg->i2c_spec, TSL2520_REG_STATUS, &reg_status);
+
+		ret = i2c_reg_read_byte_dt(&cfg->i2c_spec, TSL2520_REG_STATUS, &reg_status);
+		if (ret < 0) {
+			LOG_ERR("Failed reading chip status");
+			return -EIO;
+		}
+		ret = i2c_reg_read_byte_dt(&cfg->i2c_spec, TSL2520_REG_ALS_DATA0_LOW, &read_low_byte);
```

Comment:
i2c_burst_read_dt()

Reply:
hello @marcusfolkesson: Thank you very much for your review, but here I have some different thoughts, there are 2 independent regesisters: TSL2520_REG_ALS_DATA0_LOW and TSL2520_REG_ALS_DATA0_HIGH,  the address of them are different ,but the function of the "i2c_burst_read_dt()" is to _Read multiple bytes from an internal address of an I2C device. This is equivalent to: i2c_burst_read(spec->bus, spec->addr, start_addr, buf, num_bytes);_ , at the same time I also did some test to print the read values from these two registers the value are different, using the function "i2c_burst_read_dt()" and "i2c_reg_read_byte_dt".

Reply:
to help understanding, I will paste the print information which read value send from UART.
![image](https://github.com/zephyrproject-rtos/zephyr/assets/137849449/77a18de5-a842-4127-aa7a-73c43250e8e0)
the read format is LOG_DBG("burst_read%X,%X,%X",le16_buffer,read_low_byte,read_high_byte);